rascal-lsp/src/main/rascal/demo/lang/pico/LanguageServer.rsc

```diff
@@ -177,6 +179,15 @@ value picoExecutionService(removeDecl(start[Program] program, IdType toBeRemoved
     return ("result": true);
 }
 
+loc picoRenamePreparingService(Focus _:[Id id, *_]) = id.src;
+
+tuple[list[DocumentEdit], set[Message]] picoRenamingService(Focus focus, str newName) = <[changed(focus[0].src.top, [
+    replace(id.src, newName)
+    | cursor := focus[0]
+    , /Id id := focus[-1]
+    , id := cursor
+])], {}>;
```

Comment:
This is just (like most of the code in this file) a very simple, syntax-based example.